src/SFA.DAS.Apprentice.LoginService.MessageHandler/SendInvitationCommandHandler.cs

```diff
@@ -1,77 +1,50 @@
-using System;
-using System.Net.Http;
-using System.Threading.Tasks;
-using Microsoft.AspNetCore.Mvc;
-using Microsoft.Azure.WebJobs;
-using Microsoft.Azure.WebJobs.Extensions.Http;
 using Microsoft.Extensions.Logging;
-using Newtonsoft.Json;
-using SFA.DAS.Apprentice.LoginService.MessageHandler.Infrastructure.NServiceBus;
+using NServiceBus;
 using SFA.DAS.Apprentice.LoginService.MessageHandler.InvitationService;
-using SFA.DAS.Apprentice.LoginService.Messages;
-using SFA.DAS.NServiceBus.AzureFunction.Attributes;
+using SFA.DAS.Apprentice.LoginService.Messages.Commands;
+using System;
+using System.Threading.Tasks;
 
 namespace SFA.DAS.Apprentice.LoginService.MessageHandler
 {
-    public class SendInvitationCommandHandler
+    public class SendInvitationCommandHandler : IHandleMessages<SendInvitation>
     {
         private readonly IInvitationApi _api;
+        private readonly ILogger _log;
 
-        public SendInvitationCommandHandler(IInvitationApi api)
+        public SendInvitationCommandHandler(IInvitationApi api, ILogger log)
         {
             _api = api;
+            _log = log;
         }
 
-        [FunctionName("HandleSendInvitationCommand")]
-        public async Task RunCommand(
-            [NServiceBusTrigger(Endpoint = QueueNames.SendInvitationCommand)] SendInvitationCommand sendInvitationCommand,
-            ILogger log)
+        public async Task Handle(SendInvitation message, IMessageHandlerContext context)
         {
-            try
+            var response = await _api.SendInvitation(message.ClientId, new SendInvitationRequest
             {
-                var response = await _api.SendInvitation(sendInvitationCommand.ClientId, new SendInvitationRequest
-                {
-                    Email = sendInvitationCommand.Email,
-                    GivenName = sendInvitationCommand.GivenName,
-                    FamilyName = sendInvitationCommand.FamilyName,
-                    SourceId = sendInvitationCommand.SourceId,
-                    Callback = new Uri(sendInvitationCommand.Callback),
-                    UserRedirect = new Uri(sendInvitationCommand.UserRedirect),
-                    OrganisationName = sendInvitationCommand.OrganisationName,
-                    ApprenticeshipName = sendInvitationCommand.ApprenticeshipName,
-                    Inviter = null
-                });
+                Email = message.Email,
+                GivenName = message.GivenName,
+                FamilyName = message.FamilyName,
+                SourceId = message.SourceId,
+                Callback = new Uri(message.Callback),
+                UserRedirect = new Uri(message.UserRedirect),
+                OrganisationName = message.OrganisationName,
+                ApprenticeshipName = message.ApprenticeshipName,
+                Inviter = null
+            });
```

Comment:
I thought we were getting rid of the API layer because it wasn't authenticated, and I thought we would call the application layer directly. But this might cause other problems

Reply:
I also like the idea of using the application layer directly from within the function app.  As well as removing the security concern it seems to me it's a good idea to side-step a whole layer of infrastructure that isn't necessary.

As discussed, let's leave this PR as is and open a ticket to do that.